### PR TITLE
Throughput-optimized Mamba SSU MTP kernel (Blackwell)

### PR DESCRIPTION
**Pull request by ishovkun**
## 📌 Description
This adds a vertical Mamba SSU (decode) MTP kernel for `selective_state_update` targeting Blackwell (SM100+). This kernel processes multiple heads per CTA by exploiting the shared B/C structure across KV groups, using TMA bulk loads and 16-warp CTAs with 3 independent compute groups.

### Key changes

- **New vertical MTP kernel** (`kernel_selective_state_update_mtp_vertical.cuh`): 16-warp CTA with 3 compute groups (4 warps each), 3 TMA load warps, and 1 epilogue warp. Each group processes a full `DIM x DSTATE` head tile, amortizing B/C loads across heads in the same KV group. State writes go directly to gmem via vectorized STG (no smem intermediate).

- **File restructure**: Split the monolithic `kernel_selective_state_update_mtp.cuh` into:
  - `invoke_selective_state_update_mtp.cuh` — launcher with auto algorithm selection and alignment checks
  - `kernel_selective_state_update_mtp_simple.cuh` — the existing simple kernel
  - `kernel_selective_state_update_mtp_vertical.cuh` — the new vertical kernel

- **SM100 JIT module**: Added `gen_selective_state_update_sm100_module` with `FLASHINFER_MAMBA_ENABLE_SM100` flag. The vertical kernel requires 16 warps/block (512 threads), only supported on Blackwell+. Python API auto-selects the sm100 module on SM >= 10.

- **Auto algorithm selection**: On SM100+, `algorithm="auto"` picks vertical for `batch >= 32` (unless scaleState is active), simple otherwise. Pre-SM100 always uses simple.

- **Stochastic rounding**: Implemented in the vertical kernel using `cvt_rs_f16x2_f32` (Philox-4x32 PRNG). Packs two fp32→fp16 conversions per instruction.

- **Comprehensive test coverage**: Added vertical test classes for all feature combinations — base correctness, intermediate states, disable_state_update, non-contiguous state, int32 indices, various ngroups (1-64), large batch (up to 256), stochastic rounding, and scaleState rejection.

- **Alignment checks**: Added output and intermediate_states pointer alignment validation in the launcher.

- **Benchmark**: Added `--dtype` flag and `flashinfer_auto` backend to `bench_ssu_sweep_mtp.py`.

Note that I added an "auto" kernel type argument for the MTP case that is analogous to the single-token case. The improvements added by this kernel are show in the image below:
ssu_mtp_11

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [x] Tests have been added or updated as needed.
- [x] All tests are passing (`unittest`, etc.).

## Summary by CodeRabbit

* **New Features**
  * SM100 GPU support and a new vertical MTP execution path for selective state updates.
  * New benchmarking tool to compare kernel performance across batch sizes and dtypes with CLI and saved plots.

* **Tests**
  * Expanded coverage for vertical-path scenarios, intermediate states, dtype/scale edge cases, non-contiguous caches, and large batches.

* **Chores**
  * Reorganized MTP kernel surface and headers to expose vertical/simple execution paths.

* **Revert**
  * Removed the repository issue-claim CI workflow.

### CODE REVIEW

File: include/flashinfer/mamba/invoke_selective_state_update_mtp.cuh [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+template <typename input_t, typename weight_t, typename matrixA_t, typename state_t,
+          typename stateIndex_t, typename state_scale_t>
+void invokeSelectiveStateUpdateMTP(SelectiveStateMTPParams& params, SSUAlgorithm algorithm,
+                                   cudaStream_t stream) {
+  constexpr bool scaleState = !std::is_same_v<state_scale_t, void>;
+  // Stochastic rounding is only implemented for fp16 state
+  if constexpr (PHILOX_ROUNDS > 0) {
+    static_assert(std::is_same_v<state_t, half>,
+                  "Stochastic rounding (PHILOX_ROUNDS > 0) only supports fp16 state");
+  }
```
[the remark this answers is not in the document]
Comment (ishovkun) [reply include/flashinfer/mamba/invoke_selective_state_update_mtp.cuh:33]: Nah, it's fine. It's easier to keep track of the restriction closer to the place where it is used.

File: flashinfer/jit/mamba/selective_state_update.py [hunk 2]
```diff
@@ -204,7 +204,59 @@ def gen_selective_state_update_sm90_module(
     nvcc_flags = compilation_context.get_nvcc_flags_list(
         supported_major_versions=[9, 10, 11, 12]
     )
-    nvcc_flags += ["-DFLASHINFER_MAMBA_ENABLE_SM90"]
+    nvcc_flags += ["-DFLASHINFER_MAMBA_ENABLE_SM90", "-lineinfo"]
```
[the remark this answers is not in the document]
Comment (ishovkun) [reply flashinfer/jit/mamba/selective_state_update.py]: fixed

File: flashinfer/jit/mamba/selective_state_update.py [hunk 3]
```diff
@@ -204,7 +204,59 @@ def gen_selective_state_update_sm90_module(
...
+    )
+    compilation_context = CompilationContext()
+    nvcc_flags = compilation_context.get_nvcc_flags_list(
+        supported_major_versions=[10, 11, 12]
+    )
+    nvcc_flags += [
+        "-DFLASHINFER_MAMBA_ENABLE_SM90",
+        "-DFLASHINFER_MAMBA_ENABLE_SM100",
+        "-lineinfo",
+    ]
```
[the remark this answers is not in the document]
Comment (ishovkun) [reply flashinfer/jit/mamba/selective_state_update.py]: fixed

File: .github/workflows/issue-claim.yml
Comment (bkryu) [inline .github/workflows/issue-claim.yml:1]: Not familiar enough to review the PR, but deleting this file seems like a mistake? Can you check and fix?

Comment (ishovkun) [reply .github/workflows/issue-claim.yml:1]: Restored in [a10a843](https://github.com/flashinfer-ai/flashinfer/pull/2773/commits/a10a843fb9a88e4a2fcd26e52b727d58a1f53eb7)

File: flashinfer/jit/mamba/selective_state_update.py [hunk 4]
```diff
@@ -169,7 +169,6 @@ def gen_selective_state_update_module(
         dstate,
         ntokens_mtp,
         philox_rounds=philox_rounds,
-        extra_cuda_cflags=["-lineinfo"],
```
Comment (yzh119) [inline flashinfer/jit/mamba/selective_state_update.py:185]: FYI we have an environment variable `FLASHINFER_JIT_LINEINFO` to control whether to add this flag or not :)

Comment (ishovkun) [reply flashinfer/jit/mamba/selective_state_update.py:185]: I know but never got it working for some reason :-(
